src/identify_client.cpp

```diff
@@ -71,7 +71,9 @@ namespace {
 		if (id[0] != '-' || !is_print(char(id[1])) || (id[2] < '0')
 			|| (id[3] < '0') || (id[4] < '0')
 			|| (id[5] < '0') || (id[6] < '0')
-			|| id[7] != '-')
+			|| (id[7] != '-'
+				&& !std::equal(id.begin() + 1, id.begin() + 3, "FG")
+				&& !std::equal(id.begin() + 1, id.begin() + 3, "SP")))
```

Comment:
it doesn't quite seem appropriate to have this exception here. If people are sloppy with how they encode the Azureus-style, perhaps this function should just be more forgiving about it in general

Reply:
Ok, "SP" should be moved, not "FG", I will fix it later.

Reply:
why does "FG" belong here?

Reply:
I am not good at English, it means this commit, not the next fix.